pkg/storaged/mdraid/mdraid.jsx

```diff
@@ -0,0 +1,306 @@
+        else
+            return wipe_members();
+    }
+
+    const usage = get_active_usage(client, block ? block.path : "", _("delete"));
+
+    if (usage.Blocking) {
+        dialog_open({
+            Title: cockpit.format(_("$0 is in use"), mdraid_name(mdraid)),
+            Body: BlockingMessage(usage)
+        });
+        return;
+    }
+
+    dialog_open({
+        Title: cockpit.format(_("Permanently delete $0?"), mdraid_name(mdraid)),
+        Teardown: TeardownMessage(usage),
+        Action: {
+            Title: _("Delete"),
+            Danger: _("Deleting erases all data on a MDRAID device."),
+            action: async function () {
+                await teardown_active_usage(client, usage);
+                await delete_();
+                navigate_away_from_card(card);
+            }
+        },
+        Inits: [
+            init_active_usage_processes(client, usage)
+        ]
+    });
+}
+
+function start_stop_action(mdraid) {
+    let running = mdraid.Running;
+    if (running === undefined)
+        running = mdraid.ActiveDevices && mdraid.ActiveDevices.length > 0;
+
+    // "Stop" is only in the card, to discourage stopping.
+
+    if (!running)
+        return { title: _("Start"), action: () => mdraid_start(mdraid), tag: "device" };
+    else
+        return null;
+}
+
+function add_disk(mdraid) {
+    function filter_inside_mdraid(spc) {
+        let block = spc.block;
+        if (client.blocks_part[block.path])
+            block = client.blocks[client.blocks_part[block.path].Table];
+        return block && block.MDRaid != mdraid.path;
+    }
+
+    function rescan(path) {
+        // mdraid often forgets to trigger udev, let's do it explicitly
+        return client.wait_for(() => client.blocks[path]).then(block => block.Rescan({ }));
+    }
+
+    dialog_open({
+        Title: _("Add disks"),
+        Fields: [
+            SelectSpaces("disks", _("Disks"),
+                         {
+                             empty_warning: _("No disks are available."),
+                             validate: function (disks) {
+                                 if (disks.length === 0)
+                                     return _("At least one disk is needed.");
+                             },
+                             spaces: get_available_spaces(client).filter(filter_inside_mdraid)
+                         })
+        ],
+        Action: {
+            Title: _("Add"),
+            action: function(vals) {
+                return prepare_available_spaces(client, vals.disks).then(paths =>
+                    Promise.all(paths.map(p => mdraid.AddDevice(p, {}).then(() => rescan(p)))));
+            }
+        }
+    });
+}
+
+function missing_bitmap(mdraid) {
+    return (mdraid.Level != "raid0" &&
+            client.mdraids_members[mdraid.path].some(m => m.Size > 100 * 1024 * 1024 * 1024) &&
+            mdraid.BitmapLocation && decode_filename(mdraid.BitmapLocation) == "none");
+}
```

Comment:
This isn't just moved code, but new. What's the magic thing about 100 GB? That deserves a comment.

I suppose this goes with the "This MDRAID device has no write-intent bitmap" UI warning below.